src/sentry/middleware/staff.py

```diff
@@ -0,0 +1,57 @@
+from django.conf import settings
+from django.utils.deprecation import MiddlewareMixin
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.auth.staff import Staff, logger
+
+
+class StaffMiddleware(MiddlewareMixin):
+    def process_request(self, request: Request):
+        # This avoids touching user session, which means we avoid
+        # setting `Vary: Cookie` as a response header which will
+        # break HTTP caching entirely.
+        self.__skip_caching = request.path_info.startswith(settings.ANONYMOUS_STATIC_PREFIXES)
+
+        if self.__skip_caching:
+            return
+
+        stf = Staff(request)
+
+        request.staff = stf
+        request.is_staff = lambda: request.staff.is_active
+
+        if stf.is_active:
+            logger.info(
+                "staff.request",
+                extra={
+                    "url": request.build_absolute_uri(),
+                    "method": request.method,
+                    "ip_address": request.META["REMOTE_ADDR"],
+                    "user_id": request.user.id,
+                },
+            )
+
+    def process_response(self, request: Request, response: Response) -> Response:
+        try:
+            if self.__skip_caching:
+                return response
+        except AttributeError:
+            pass
+        stf = getattr(request, "staff", None)
+        if stf:
+            if stf.is_active:
+                # What do we log here? Should we log access to orgs?
+                org_slug = getattr(getattr(request, "organization", None), "slug", None)
+                if org_slug:
+                    logger.info(
```

Comment:
hmm... do we need to log anything here? any changes made to customer accounts in `_admin` should be audit logged

Reply:
should prob be fine to skip logging